src/wallet/feebumper.cpp

```diff
@@ -388,5 +388,102 @@ Result CommitTransaction(CWallet& wallet, const uint256& txid, CMutableTransacti
     return Result::OK;
 }
 
+Result CreateRateBumpDeniabilizationTransaction(CWallet& wallet, const uint256& txid, unsigned int confirm_target, bool sign, bilingual_str& error, CAmount& old_fee, CAmount& new_fee, CTransactionRef& new_tx)
```

Comment:
```suggestion
Result CreateRateBumpDeniabilizationTransaction(
        CWallet& wallet, const uint256& txid, unsigned int confirm_target,
        bool sign, bilingual_str& error, CAmount& old_fee, CAmount& new_fee,
        CTransactionRef& new_tx)
```
so many params, did you think of removing some?

Reply:
CreateRateBumpDeniabilizationTransaction essentially does the job of CreateRateBumpTransaction and SignTransaction, but specialized for bumping deniabilization transactions. 
This functionality is responsible for the presence of the "txid", "old_fee", "new_fee" and "sign" parameters.
"confirm_target" takes the place of the "coin_control" parameter (since that's all we need), and the "error" and "new_tx" are needed to communicate the result/error with the caller.
The reason we need a new API function here is that we need to take the new bump fee from one of the recipients of the old transaction (fSubtractFeeFromAmount), and there didn't seem to be a way to do that via CreateRateBumpTransaction. And generally CreateRateBumpTransaction just does too many other things that we don't need once we know we're bumping a deniabilization transaction.